lib/src/deploy.rs

```diff
@@ -278,20 +278,21 @@ async fn deploy(
     image: &ImageState,
     origin: &glib::KeyFile,
     opts: Option<ostree::SysrootDeployTreeOpts<'_>>,
-) -> Result<()> {
+) -> Result<Deployment> {
```

Comment:
We can split this out as its own prep PR that can land immediately.

Reply:
WDYT about doing that now? I think we could also add the code using `openat` to read the target root.

Reply:
I also did https://github.com/containers/bootc/pull/674/commits/009c08adb4f427cdcde2add48f5683817ef9c19f#diff-b28d7b8052d2d63505b324be5a3affca036ff7060cba6fc61f4519e29ca5ca9eR35 while looking at code related to this for a different reason...although - and this is subtle - today [this readme describes why cap-std doesn't use RESOLVE_IN_ROOT](https://github.com/bytecodealliance/cap-std/?tab=readme-ov-file#similar-crates) ...so we'd need to add a variant that looks like https://github.com/containers/composefs/pull/286/files#diff-2c1a4b8055c87401bf768c30392660303de5e336af37c1cec34e3eeaee55e6deR85 that drops down to the raw rustix apis.

(In general for us, it actually would be nice potentially to add to cap-std a variant that uses `RESOLVE_IN_ROOT`...I'll chime in on https://github.com/bytecodealliance/cap-std/issues/353 about that)

Reply:
just opened #675 to address the first part. I'm not quite sure where to put the openat code other than in the BoundImage functions of this PR.

I also just pushed my latest WIP changes to this PR's branch. I've done some initial testing of the latest changes, it seems to be basically working. I still need to add concurrency to the pulls, tests, docs, etc. but I wanted to get the current state out there for an initial review.